src/main/java/core/mate/academy/service/MachineServiceImpl.java

```diff
@@ -1,7 +1,42 @@
 package core.mate.academy.service;
 
+import core.mate.academy.model.Bulldozer;
+import core.mate.academy.model.BulldozerProducer;
+import core.mate.academy.model.Excavator;
+import core.mate.academy.model.ExcavatorProducer;
+import core.mate.academy.model.Machine;
+import core.mate.academy.model.Truck;
+import core.mate.academy.model.TruckProducer;
+import java.util.ArrayList;
+import java.util.List;
+
 /**
  * Your implementation of MachineService.
  */
-public class MachineServiceImpl {
+public class MachineServiceImpl implements MachineService<Machine> {
+    @Override
+    public List<Machine> getAll(Class<? extends Machine> type) {
+        if (type.equals(Truck.class)) {
+            return new TruckProducer().get();
+        } else if (type.equals(Bulldozer.class)) {
```

Comment:
```suggestion
        } 
        if (type.equals(Bulldozer.class)) {
```
нам не потрібню юзати else якщо ми в if робимо return